phoenix-core-client/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -5052,6 +5070,39 @@ void ensureSystemTablesMigratedToSystemNamespace()
         }
     }
 
+    /**
+     * Acquire distributed mutex of sorts to make sure only one JVM is able to run the upgrade code by
```

Comment:
That's not really what this mutex does.
The single JVM problem is handled by acquireUpgradeMutex().

This one is just blocking all clients which have not explicitly overriden this.

Reply:
This comment is still misleading.

Reply:
This comment is still incorrect.
While we do use the mutex table, we do do not really use it as mutex in this case, we do not use to make sure that only one JVM is able to upgrade (at least not directly), and we do not use the checkAndPut API.